include/swift/SIL/SILFunction.h

```diff
@@ -1637,6 +1635,17 @@ class SILFunction
         decl->getLifetimeAnnotation());
   }
 
+  ArrayRef<std::pair<SILType, SILUndef *>> getUndefValues() {
+    return {undefValues.begin(), undefValues.end()};
+  }
+
+  void updateTypeForUndef(SILUndef *undef, SILType oldType) {
```

Comment:
It's strange for an API called "updateType" to be passed the old value for the type. It makes more sense to pass the old and new undef values instead and call it "replaceUndef", or pass the old and new types and call it "replaceUndefType"

I can't make a suggestion beyond that because I don't understand where and how the undef's type actually changes.

Reply:
> or pass the old and new types and call it "replaceUndefType"

Fixed.

> where and how the undef's type actually changes

The same way as every other value this pass modifies: `unsafelyEliminateMoveOnlyWrapper`.

Reply:
> The same way as every other value this pass modifies: unsafelyEliminateMoveOnlyWrapper.

Ok, looking at this one commit at a time, it wasn't clear that `recordValue(pair.second)` actually updates types. It sounds like it's actually recording the value for later update.

Reply:
Changed the name to `visitValue` and added a short comment.  Also, used `RAUW` for undefs because there might already be an undef of the unwrapped type in the function's undef map.